### PR TITLE
Preparation for release v2023.1.4 (Zonemaster/Zonemaster)

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

Preparation for release.

## Changes

Updates files `Changes` and `zonemaster-gui.md` (installation).

## How to test this PR

Review the changes.

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+ [Zonemaster product]
+ - This version of Zonemaster also consists of the following components. For
+   each component, see its Changes file or Github release notes for complete
+   release information.
+
+   Zonemaster-LDNS      v3.2.0  https://github.com/zonemaster/zonemaster-ldns/releases/tag/v3.2.0
+   Zonemaster-Engine    v4.7.2  https://github.com/zonemaster/zonemaster-engine/releases/tag/v4.7.3
+   Zonemaster-CLI       v6.0.2  https://github.com/zonemaster/zonemaster-cli/releases/tag/v6.0.3
+   Zonemaster-Backend  v11.0.1  https://github.com/zonemaster/zonemaster-backend/releases/tag/v11.0.2
+   Zonemaster-GUI       v4.1.0  https://github.com/zonemaster/zonemaster-gui/releases/tag/v4.1.1
```
Comment (mattias-p) [inline Changes:21]:
```suggestion
   Zonemaster-Engine    v4.7.3  https://github.com/zonemaster/zonemaster-engine/releases/tag/v4.7.3
   Zonemaster-CLI       v6.0.3  https://github.com/zonemaster/zonemaster-cli/releases/tag/v6.0.3
   Zonemaster-Backend  v11.0.2  https://github.com/zonemaster/zonemaster-backend/releases/tag/v11.0.2
   Zonemaster-GUI       v4.1.1  https://github.com/zonemaster/zonemaster-gui/releases/tag/v4.1.1
```

Comment (matsduf) [reply Changes]: Good catch!

**Review by mattias-p — APPROVED** [review]

File: Changes [hunk 2]
```diff
@@ -1,5 +1,26 @@
 Release history of Zonemaster
 
+v2023.1.4 2023-09-05 (public fix version)
+
+ [Fixes]
+ - Fixes a bug (regression) in test case BASIC01 that made tests of zones
+   fail if the delegation to the zone had no glue record, which is as
+   expected if the zone has no in-bailiwick NS. The bug was introduced
+   in version v4.7.2 in release v2023.1.2. See Zonemaster-Engine.
```
Comment (marc-vanderwal) [inline Changes:9]: The failure only happens when a referral points to out-of-bailiwick name servers for a zone that isn’t the zone to test, but some parent of it.

Firstly, `ip6.arpa` does not exhibit the problem because the first delegation is from the root to `arpa`, which has in-bailiwick name servers with glue, then to `ip6.arpa`—the zone we want to test.

Secondly, `7.0.1.0.0.2.ip6.arpa` also does not exhibit the problem, because the delegation from `arpa` to `ip6.arpa` has out-of-bailiwick name servers but contains glue records. Then, there is a delegation from `ip6.arpa` to `7.0.1.0.0.2.ip6.arpa`—the zone we want to test.

Finally, `0.0.9.0.0.0.7.0.1.0.0.2.ip6.arpa` did exhibit the problem. That’s because the delegation from `ip6.arpa` to `7.0.1.0.0.2.ip6.arpa`, which is a parent of the zone to test, has out-of-bailiwick name servers without glue records. The code then failed to obtain IPv4 and IPv6 addresses for the listed name servers.

In short, the presence or lack of glue was not a problem if the referral pointed to the zone under test. If not, then the bug happened.

```suggestion
 - Fixes a bug (regression) in test case BASIC01 that made tests of zones fail
   if, while walking the chain of delegations from the root to the zone under
   test, a name server returns a referral pointing to name servers for an
   intermediate zone which are all out-of-bailiwick and lacking glue. The bug
   was introduced in version v4.7.2 in release v2023.1.2. See
   Zonemaster-Engine.
```

**Review by marc-vanderwal — COMMENTED** [review]
Just one suggestion, as requested, for the description of the only bug fix in this release.

**Comment by matsduf** [comment]
@marc-vanderwal, thank you for the description. I think it rather belongs to https://github.com/zonemaster/zonemaster-engine/pull/1282

**Comment by matsduf** [comment]
@mattias-p, @marc-vanderwal and @tgreenx, please (re-) review this PR, especially the updated Changes file.

**Review by marc-vanderwal — APPROVED** [review]

**Comment by marc-vanderwal** [comment]
> @marc-vanderwal, thank you for the description. I think it rather belongs to [zonemaster/zonemaster-engine#1282](https://github.com/zonemaster/zonemaster-engine/pull/1282)

Yes indeed. I’ve added a similar comment in the PR.

**Review by mattias-p — APPROVED** [review]

**Review by marc-vanderwal — APPROVED** [review]

**Review by marc-vanderwal — APPROVED** [review]
